Adds test suite for word-count

m-a-ge opened the pull request:
Resolves #77 

I've ported the tests from [`canonical-data.json`](https://github.com/exercism/problem-specifications/blob/master/exercises/word-count/canonical-data.json) for the exercise, but I didn't have a chance to run them locally.

I've never written unit tests for bash scripts, so there could be mistakes. I'll check everything tomorrow

Feedback is always welcomed! 🤝

kotp commented on exercises/word-count/word_count_test.sh:
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env bats
+
+@test "count one word" {
+    run ./word-count.sh "word"
```
Should the name of the `word-count.sh` match the naming convention of the tests file, with underlines rather than dashes?  Right now, the file name for the run commands is with a dash.

m-a-ge replied:
Agreed. I'll change it, thanks!

m-a-ge commented:
PR was updated

I've finally set up `bats` on my Ubuntu machine, ran the suite and faced several issues:
1. command `run ./word_count.sh "test"` in tests didn't work, but ` run bash word_count.sh "one of each"` succeeded 
2. order of output matters, don't know how to implement tests to ignore it
3. existing implementation doesn't pass all the added tests from `canonical-data.json`, I've commented  them out

Not sure how to resolve the issues mentioned above

kotp commented:
To solve point 3, write the example solution that conforms to the tests.

Can you show your run command and results for point 1?

Point 2...  perhaps solving point 3 will resolve this.

m-a-ge commented:
Ok, updated the PR and figured out how to check if output contains required string.
I've also fixed example implementation to pass additional tests. 
Unfortunately I didn't manage to write a solution which could pass the last test (string with apostrophe and word in single quotes). This is quite difficult.

TODO:
1. In tests check that output contains not only all required tests but number of outputted lines is also correct (no extra lines are shown)
1. Implement solution for the last test

kytrinyx commented:
> Unfortunately I didn't manage to write a solution which could pass the last test (string with apostrophe and word in single quotes). This is quite difficult.

It would be fine to remove that test for this track. I don't want people to have to wrestle with it unless there's a fairly straight-forward solution in the language.

m-a-ge commented:
I've finally finished with my PR.
Sorry for the late response, but I broke my right arm yesterday and it's very hard to type with one hand :disappointed:

@Smarticles101, thanks for the approval
@exercism/bash, the PR is ready for merge, I've squashed my commits
I took @kytrinyx's advice and didn't include the last test from `canonical-data.json` which deals with apostrophes, but I've added additional verification to the test suite that test how many words is printed. 

Please note that this PR is created to `word-count` branch and it's required to merge it to master after this one.

kytrinyx commented:
Oh, no, @m-a-ge, I hope you get better quickly! 🏥

kytrinyx commented:
I'm going to go ahead and let @kotp merge this one, so he can verify that his previous comments have been addressed (it all looks good to me!)

kotp commented:
Looking good @m-a-ge and merged!  Get well soon!
